transport/http/outbound.go

```diff
@@ -445,6 +400,118 @@ func getYARPCErrorFromResponse(response *http.Response, bothResponseError bool)
 	)
 }
 
+// RoundTrip implements the http.RoundTripper interface, making a YARPC HTTP outbound suitable as a
+// Transport when constructing an HTTP Client. An HTTP client is suitable only for relative paths to
+// a single outbound service. The HTTP outbound overrides the host:port portion of the URL of the
+// provided request.
+//
+// Sample usage:
+//
+//  client := http.Client{Transport: outbound}
+//
+// Thereafter use the Golang standard library HTTP to send requests with this client.
+//
+//  ctx := context.Background()
+//  ctx, cancel := context.WithTimeout(ctx, time.Second)
+//  defer cancel()
+//  req := http.NewRequest("GET", "http://example.com/", nil)
+//  req = req.WithContext(ctx)
+//  res, err := client.Do(req)
+//
+// All requests must have a deadline on the context.
+// The peer chooser for raw HTTP requests will receive a blank YARPC transport.Request, which is
+// sufficient for load balancers like peer/pendingheap (fewest-pending-requests) and peer/roundrobin
+// (round-robin).
+func (o *Outbound) RoundTrip(hreq *http.Request) (*http.Response, error) {
+	return o.roundTrip(hreq, nil /* treq */, time.Now())
+}
+
+func (o *Outbound) roundTrip(hreq *http.Request, treq *transport.Request, start time.Time) (*http.Response, error) {
+	ctx := hreq.Context()
+
+	deadline, ok := ctx.Deadline()
```

Comment:
`call` also calculates `ttl` before calling this method, any reason we do it twice (And note that it the `ttl` may change between calls)

Reply:
start and deadline are fixed, so the ttl is the same. The alternative approach is to duplicate the code to compute the TTL in RoundTrip to thread into roundTrip as an argument.